config.m4

```diff
@@ -197,52 +208,66 @@ if test "$PHP_MONGODB" != "no"; then
-    m4_include([scripts/build/autotools/m4/ac_compile_check_sizeof.m4])
-    m4_include([scripts/build/autotools/m4/ac_create_stdint_h.m4])
-    m4_include([scripts/build/autotools/m4/as_var_copy.m4])
-    m4_include([scripts/build/autotools/m4/ax_check_compile_flag.m4])
-    m4_include([scripts/build/autotools/m4/ax_prototype.m4])
-    m4_include([scripts/build/autotools/m4/ax_pthread.m4])
-    m4_include([scripts/build/autotools/m4/pkg.m4])
-
-    m4_include([scripts/build/autotools/CheckCompiler.m4])
-    m4_include([scripts/build/autotools/CheckHost.m4])
-
-    m4_include([scripts/build/autotools/libbson/CheckAtomics.m4])
-    m4_include([scripts/build/autotools/libbson/CheckHeaders.m4])
-    m4_include([scripts/build/autotools/libbson/Endian.m4])
-    m4_include([scripts/build/autotools/libbson/FindDependencies.m4])
-    m4_include([scripts/build/autotools/libbson/Versions.m4])
-
-    m4_include([scripts/build/autotools/libmongoc/CheckCompression.m4])
-    m4_include([scripts/build/autotools/libmongoc/CheckResolv.m4])
-    m4_include([scripts/build/autotools/libmongoc/CheckSasl.m4])
-    m4_include([scripts/build/autotools/libmongoc/CheckSSL.m4])
-    m4_include([scripts/build/autotools/libmongoc/FindDependencies.m4])
-    m4_include([scripts/build/autotools/libmongoc/PlatformFlags.m4])
-    m4_include([scripts/build/autotools/libmongoc/Versions.m4])
-    m4_include([scripts/build/autotools/libmongoc/WeakSymbols.m4])
+    _include([scripts/build/autotools/m4/ac_compile_check_sizeof.m4])
+    _include([scripts/build/autotools/m4/ac_create_stdint_h.m4])
+    _include([scripts/build/autotools/m4/as_var_copy.m4])
+    _include([scripts/build/autotools/m4/ax_check_compile_flag.m4])
+    _include([scripts/build/autotools/m4/ax_prototype.m4])
+    _include([scripts/build/autotools/m4/ax_pthread.m4])
+    _include([scripts/build/autotools/m4/php_mongodb.m4])
+    _include([scripts/build/autotools/m4/pkg.m4])
+
+    _include([scripts/build/autotools/CheckCompiler.m4])
+    _include([scripts/build/autotools/CheckHost.m4])
+
+    _include([scripts/build/autotools/libbson/CheckAtomics.m4])
+    _include([scripts/build/autotools/libbson/CheckHeaders.m4])
+    _include([scripts/build/autotools/libbson/Endian.m4])
+    _include([scripts/build/autotools/libbson/FindDependencies.m4])
+    _include([scripts/build/autotools/libbson/Versions.m4])
+
+    _include([scripts/build/autotools/libmongoc/CheckCompression.m4])
+    _include([scripts/build/autotools/libmongoc/CheckResolv.m4])
+    _include([scripts/build/autotools/libmongoc/CheckSasl.m4])
+    _include([scripts/build/autotools/libmongoc/CheckSSL.m4])
+    _include([scripts/build/autotools/libmongoc/FindDependencies.m4])
+    _include([scripts/build/autotools/libmongoc/PlatformFlags.m4])
+    _include([scripts/build/autotools/libmongoc/Versions.m4])
+    _include([scripts/build/autotools/libmongoc/WeakSymbols.m4])
+
+    m4_popdef([_include])
```

Comment:
Currently there is no risk of conflicting with an `_include` macro defined outside of this push/pop scope. The only risk would be if one of the M4 files we include defined its own `_include` macro without also using push/pop and overwrote our own.

`PHP_MONGODB_ADD_SOURCES` and friends can't use push/pop (reasonably) since we define them in an included file and call them later on in `config.m4`. Prefixing makes sense there to be a good citizen, but I don't think prefixing `_include` would add anything here (apart from verbosity).

Reply:
OK